src/pages/api/elasticSearchProxy/sources.ts

```diff
@@ -0,0 +1,112 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { client } from "@/config/elasticsearch";
+
+// types for Elasticsearch responses
+interface DomainAggregationBucket {
+  key: string;
+  doc_count: number;
+}
+
+interface LastIndexedBucket {
+  key: string;
+  last_indexed: {
+    value: number;
+  };
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method !== "POST") {
+    return res.status(405).json({
+      error:
+        "Invalid request method. This endpoint only supports POST requests.",
+    });
+  }
+
+  try {
+    // Step 1: Get unique domains and document counts
+    const domainsResult = await client.search({
+      index: process.env.INDEX,
+      body: {
+        size: 0,
+        aggs: {
+          domains: {
+            terms: {
+              field: "domain.keyword",
+              size: 1000, // Adjust this value based on the expected number of unique domains
+            },
+          },
+        },
+      },
+    });
+
+    const domainBuckets = (
+      domainsResult.aggregations?.domains as {
+        buckets: DomainAggregationBucket[];
+      }
+    ).buckets;
+
+    const domains = domainBuckets.map((bucket) => ({
+      domain: bucket.key,
+      documentCount: bucket.doc_count,
+    }));
+
+    // Step 2: Get last indexed date for each domain
```

Comment:
This API call (lastIndexedResult) returns the domains and last_indexed value.
I can see the mapping of each domain to their last_indexed value, but since this query returns the domain as well in the response, then only this API call is needed.

Reply:
good catch, fixed!